test/unittest/reduction/reduction_test.cpp

```diff
@@ -103,14 +104,10 @@ void run_test(const combination_t<scalar_t> combi) {
       };
       break;
     case operator_t::Product:
-      reduction_func = [=](data_t l, data_t r) -> data_t {
-        return abs(l) * abs(r);
```

Comment:
We don't need the abs for product and division. This probably a typo. Can you do the same fix for the expression tests as well?
This could potentially fix the product and division errors we saw previously